examples/python_boids_spatial3D_wrapped_native/boids_spatial3D.py

```diff
@@ -281,7 +279,7 @@ def inputdata(message_in: pyflamegpu.MessageSpatial3D, message_out: pyflamegpu.M
 
 env = model.Environment()
 # Population size to generate, if no agents are loaded from disk
-env.newPropertyUInt("POPULATION_TO_GENERATE", 4000)
```

Comment:
we probably don't want to commit to 100k agents by default. Debug builds and smaller devices will suffer.

Reply:
Even a laptop GPU can handle 100k agents. I don't want all our demos to be small and unimpressive. I will however add a comment that this should be reduced for debug etc.

Reply:
The lowest compute capability mobile GPU we support (SM35) is the 920M, a 33W, 384 core, 14.4Gb/s 733 GFLOPS device. It will not deal with a 100k agent sim, even if it only invovlves spatial comms (boids flocking leads to relatively dense comms). 

That is the most extreme case, but given this boids example is intended to be a better than real time visualisation, we should care about perf on a more recent but not great GPU, not everyone has a 3080.
Realistically Pascal is the current minimum arch to care about, my old 1070 for instance was a 15SM device, so 30k resident threads maximum, but achieved ~20k at the time (curve changes may have altererd this). 

I'm not saying it should be 4000 agents, but it probably shouldn't be 100,000.

Reply:
40k runs at ~170 steps per second on my 3060ti. I've not got a laptop with a GPU to test an older / smaller device (had to enable the fps counter and change the background colour to see it). Perf does degrade as the step count progresses, as flocks form and local density increases.

100k slows to ~30sps by 3000 steps, so smaller/older devices definately wouldn't have maintained a useful rate.

Reply:
I've changed to 40k. @Robadob I have tweaked parameters to that even at high numbers "you can actually make out the model detail whilst seeing the whole population".